src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpReceiver.php

```diff
@@ -42,51 +42,72 @@ public function __construct(Connection $connection, SerializerInterface $seriali
     public function receive(callable $handler): void
     {
         while (!$this->shouldStop) {
-            $AMQPEnvelope = $this->connection->get();
+            try {
+                $AMQPEnvelope = $this->connection->get();
+            } catch (\AMQPException $exception) {
+                throw new TransportException($exception->getMessage(), 0, $exception);
+            }
+
             if (null === $AMQPEnvelope) {
-                $handler(null);
+                $handler(null, null);
 
-                usleep($this->connection->getConnectionCredentials()['loop_sleep'] ?? 200000);
+                usleep($this->connection->getConnectionConfiguration()['loop_sleep'] ?? 200000);
                 if (\function_exists('pcntl_signal_dispatch')) {
                     pcntl_signal_dispatch();
                 }
 
                 continue;
             }
 
-            try {
-                $handler($this->serializer->decode([
-                    'body' => $AMQPEnvelope->getBody(),
-                    'headers' => $AMQPEnvelope->getHeaders(),
-                ]));
-
-                $this->connection->ack($AMQPEnvelope);
-            } catch (RejectMessageExceptionInterface $e) {
-                try {
-                    $this->connection->reject($AMQPEnvelope);
-                } catch (\AMQPException $exception) {
-                    throw new TransportException($exception->getMessage(), 0, $exception);
-                }
-
-                throw $e;
-            } catch (\AMQPException $e) {
-                throw new TransportException($e->getMessage(), 0, $e);
-            } catch (\Throwable $e) {
-                try {
-                    $this->connection->nack($AMQPEnvelope, AMQP_REQUEUE);
-                } catch (\AMQPException $exception) {
-                    throw new TransportException($exception->getMessage(), 0, $exception);
-                }
+            $handler($this->serializer->decode([
```

Comment:
You still need to catch the `RejectMessageExceptionInterface` exception and reject, isn't it? It's removing a very convenient feature here. 🤔

Reply:
It's still there as a `UnrecoverableMessageHandlingException`. I had to move it because it's now outside of the `Amqp` namespace, and I also renamed it. But I can rename it back (but it should be in a new namespace either way)